src/manifolds/Tucker.jl

```diff
@@ -0,0 +1,876 @@
+    > SIAM Journal on Matrix Analysis and Applications, 21(4), pp. 1253-1278, 2000
+    > doi: [10.1137/S0895479896305696](https://doi.org/10.1137/S0895479896305696)
+
+[^Koch2010]:
+    > Othmar Koch, Christian Lubic, "Dynamical Tensor approximation"
+    > SIAM Journal on Matrix Analysis and Applications, 31(5), pp. 2360-2375, 2010
+    > doi: [10.1137/09076578X](https://doi.org/10.1137/09076578X)
+"""
+struct Tucker{N,R,D,𝔽} <: AbstractManifold{𝔽} end
+function Tucker(n⃗::NTuple{D,Int}, r⃗::NTuple{D,Int}, field::AbstractNumbers=ℝ) where {D}
+    @assert is_valid_mlrank(n⃗, r⃗)
+    return Tucker{n⃗,r⃗,D,field}()
+end
+
+#=
+HOSVD{T, D}
+
+Higher-order singular value decomposition of an order D tensor with eltype T
+fields:
+* U: singular vectors of the unfoldings
+* core: core tensor
+* σ : singular values of the unfoldings
+=#
+struct HOSVD{T,D}
+    U::NTuple{D,Matrix{T}}
+    core::Array{T,D}
+    σ::NTuple{D,Vector{T}}
+end
+
+"""
+TuckerPoint{T, D}
+
+An order D tensor of fixed multilinear rank and entries of type T. The tensor is
+represented in HOSVD form. See also [`Tucker`](@ref).
+
+# Constructors:
+    TuckerPoint(core::AbstractArray{T, D}, factors::Vararg{MtxT, D}) where {T, D, MtxT <: AbstractMatrix{T}}
+
+A tensor of the form (factors[1], …, factors[D]) ⋅ core
+where it is assumed that the dimensions of the core are the multilinear rank of the tensor.
+
+    TuckerPoint(A::AbstractArray{T, D}, mlrank::NTuple{D, Int}) where {T, D}
+
+The low-multilinear rank tensor arising from the sequentially truncated the higher-order
+singular value decomposition of A [^Vannieuwenhoven2012].
+
+[^Vannieuwenhoven2012]:
+    > Nick Vannieuwenhoven, Raf Vandebril, Karl Meerbergen: "A new truncation strategy for the higher-order singular value decomposition"
+    > SIAM Journal on Scientific Computing, 34(2), pp. 1027-1052, 2012
+    > doi: [10.1137/110836067](https://doi.org/10.1137/110836067)
+
+"""
+struct TuckerPoint{T,D} <: AbstractManifoldPoint
+    hosvd::HOSVD{T,D}
+end
+function TuckerPoint(
+    core::AbstractArray{T,D},
+    factors::Vararg{MtxT,D},
+) where {T,D,MtxT<:AbstractMatrix{T}}
+    # Take the QR decompositions of the factors and multiply the R factors into the core
+    qrfacs = qr.(factors)
+    Q = map(qrfac -> qrfac.Q, qrfacs)
+    R = map(qrfac -> qrfac.R, qrfacs)
+    core′ = reshape(Kronecker.:⊗(reverse(R)...) * vec(core), size(core))
+
+    # Convert to HOSVD format by taking the HOSVD of the core
+    decomp = st_hosvd(core′)
+    factors′ = Q .* decomp.U
+    return TuckerPoint(HOSVD{T,D}(factors′, decomp.core, decomp.σ))
+end
+function TuckerPoint(A::AbstractArray{T,D}, mlrank::NTuple{D,Int}) where {T,D}
+    @assert is_valid_mlrank(size(A), mlrank)
+    return TuckerPoint(st_hosvd(A, mlrank))
+end
+
+@doc raw"""
+    TuckerTVector{T, D} <: TVector
+
+Tangent vector to the D-th order Tucker manifold at $x = (U_1,\dots,U_D) ⋅ \mathcal{C}$.
```

Comment:
In most cases we use `p` as a point on the manifold. You seem to use usually A here, which is fine, too, but please be consistent then.

Style-wise we often prefer two backticks `` instead of $ for math in Markdown

Reply:
If you just could check for math being in `` (instead of $)? That is more markdown-y

Reply:
I only use the $ notation whenever I refer to symbols that also occur in a more complicated formula that requires TeX typesetting. This is one example, because the expression for the point is used in the equation below. In all other cases, I used backticks.

Reply:
Sure, that is reasonable and even preferable to use a math environment when the formulae are better readable that way (actually U even use it whenever I refer to math variables an not to input/output variables).

While in TeX you use `$...$` in Markdown you can also use ` ``...`` `, which is what we use mostly (we haven‘t gone through all the old `$...$`s yet. In rendered HTML (Documentation) both yield the same. So I actually really mean the double backticks here :)